contrib/segmentio/kafka.go.v0/kafka.go

```diff
@@ -191,10 +224,58 @@ func (w *Writer) WriteMessages(ctx context.Context, msgs ...kafka.Message) error
 	spans := make([]ddtrace.Span, len(msgs))
 	for i := range msgs {
 		spans[i] = w.startSpan(ctx, &msgs[i])
+		setProduceCheckpoint(w.cfg.dataStreamsEnabled, &msgs[i], w.Writer)
 	}
 	err := w.Writer.WriteMessages(ctx, msgs...)
 	for i, span := range spans {
 		finishSpan(span, msgs[i].Partition, msgs[i].Offset, err)
 	}
 	return err
 }
+
+func setProduceCheckpoint(enabled bool, msg *kafka.Message, writer *kafka.Writer) {
+	if !enabled || msg == nil {
+		return
+	}
+
+	var topic string
+	if writer.Topic != "" {
+		topic = writer.Topic
+	} else {
+		topic = msg.Topic
+	}
+
+	edges := []string{"direction:out", "topic:" + topic, "type:kafka"}
+	carrier := messageCarrier{msg}
+	ctx, ok := tracer.SetDataStreamsCheckpointWithParams(
+		datastreams.ExtractFromBase64Carrier(context.Background(), carrier),
+		options.CheckpointParams{PayloadSize: getProducerMsgSize(msg)},
+		edges...,
+	)
+	if !ok {
+		return
+	}
+
+	// Headers will be dropped if the current protocol does not support them
+	datastreams.InjectToBase64Carrier(ctx, carrier)
+}
+
+func getProducerMsgSize(msg *kafka.Message) (size int64) {
```

Comment:
There is an alternative private implementation here:

https://github.com/segmentio/kafka-go/blob/ebca72eaee918d303c532feb7ff29afdcd8c2efa/message.go#L48-L50

I've kept with the one in the other integrations.